File: Components/DevExpressVCL/Demos/ExpressPDFViewer/Delphi/Common/dxProgressDialog.pas
unit dxProgressDialog;

{$I cxVer.inc}

interface

uses
{$IFDEF DELPHI16}
  System.UITypes,
{$ENDIF}
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms, Dialogs, StdCtrls, cxGraphics,
  cxControls, cxClasses, cxLookAndFeels, cxLookAndFeelPainters, cxContainer, cxEdit, cxProgressBar, cxLabel, Menus,
  cxButtons;

type
  { TfrmProgress }

  TfrmProgress = class(TForm, IcxProgress)
    lbTitle: TcxLabel;
    pbProgress: TcxProgressBar;
  private
    // IcxProgress
    procedure OnProgress(Sender: TObject; Percent: Integer);
  end;

implementation

{$R *.dfm}

{ TfrmProgress }

procedure TfrmProgress.OnProgress(Sender: TObject; Percent: Integer);
begin
  pbProgress.Position := Percent;
  Application.ProcessMessages;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressPDFViewer/CBuilder/Common/dxProgressDialog.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "dxProgressDialog.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxControls"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxEdit"
#pragma link "cxLabel"
#pragma link "cxProgressBar"
#pragma resource "*.dfm"
TfrmProgress *frmProgress;
//---------------------------------------------------------------------------
__fastcall TfrmProgress::TfrmProgress(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
